fritz_interview.py

```diff
@@ -0,0 +1,70 @@
+# Lucky Dollar Store Coding Challenge
+import random
+names = ["Eric", "Chris", "Dan", "Jameson"]
+		
+
+def record_customer_purchase(customer):
+     # Fill this in
+     print("- Added new raffle ticket for \"%s\"" % customer)
+     return customer
+
+
+def can_win(customer, pool):
+	if not pool:  
+		return True
+	elif customer in pool.keys():
+		if pool[customer] == 0:
+			return True
+	elif customer not in pool.keys():
+		return True
+	return False
+
+
+def run_weekly_raffle(num_winners, num_purchases):
+	# pick winner(s)
+	pool = []
+	winning_customer = []
+	for n in range(0, num_purchases):
+		customer_index = random.randrange(0, len(names))
+		pool.append(record_customer_purchase(names[customer_index]))
+	
+	for i in range(0, num_winners):
+		winner_index = random.randrange(0, len(pool))
+		winner = pool[winner_index]
+		winning_customer.append(winner)
+	return winning_customer
+
+def main():
+	pool = []
+	num_drawings = 6
+	cant_win_again = {}
+	for i in range(1, num_drawings + 1):	
+		print("Raffle Drawing %d" % i)
+		num_purchases = random.randrange(1,5)
+		winners = list(set(run_weekly_raffle(2, num_purchases))) # same winner can't win in same drawing
+		print("- Picking the weekly winner….", end = " ")
+		# how many winners to print
+		if len(winners) == 1 and can_win(winners[0], cant_win_again):
```

Comment:
I deleted the if clause that checked the length of the list and am now checking only if the list is empty of not. If the list of winners is empty, then no one wins. Otherwise, we print each winner and update the number of raffles they have left until they can win again.